src/zcash/ProvingServiceClient.cpp

```diff
@@ -19,6 +19,26 @@ ProvingServiceClient::ProvingServiceClient(std::string id, char* url, char* serv
     assert(zmq_curve_keypair(zmqPublicKey, zmqSecretKey) == 0);
 }
 
+void ProvingServiceClient::configure_from_options()
+{
+    zmqUrl = GetArg("-provingservice", "");
+
+    // Parameters are checked in init.cpp
+    std::memcpy(zmqServerKey, GetArg("-provingservicepubkey", "").c_str(), 41);
```

Comment:
Just spotted a bug: this will copy data from memory if not set (which will never be used, but is still not good). I'll fix this, and since I'm going back into the code I'll add the gate around DP-Z.